packages/mds-agency/agency-candidate-request-handlers.ts

```diff
@@ -34,16 +34,19 @@ export const readAllVehicleIds = async (req: AgencyApiRequest, res: AgencyApiRes
   const items = await db.readDeviceIds(query_provider_id)
   const data: { [s: string]: string[] } = {}
   const summary: { [s: string]: number } = {}
-  items.map(item => {
-    const { device_id, provider_id } = item
-    if (data[provider_id]) {
-      data[provider_id].push(device_id)
-      summary[providerName(provider_id)] += 1
-    } else {
-      data[provider_id] = [device_id]
-      summary[providerName(provider_id)] = 1
-    }
-  })
+  await Promise.all(
+    items.map(async item => {
+      const { device_id, provider_id } = item
+      const name = await providerName(provider_id)
```

Comment:
It might be preferable to await getProviders and get them all outside the .map function.